File: packages/js/core/src/algorithms/combine-paths.ts
export const combinePaths = (a: string, b: string): string => {
  // Normalize all path separators
  a = a.replace(/\\/g, "/");
  b = b.replace(/\\/g, "/");

  // Append a separator if one doesn't exist
  if (a[a.length - 1] !== "/") {
    a += "/";
  }

  // Remove any leading separators from
  while (b[0] === "/" || b[0] === ".") {
    b = b.substring(1);
  }

  return a + b;
};

File: packages/py/core/src/algorithms/combine_paths.py
def combine_paths(a: str, b: str) -> str:
    # Normalize all path separators
    a = a.replace("/\\/g", "/")
    b = b.replace("/\\/g", "/")

    if a[-1] != "/":
        a += "/"

    # Remove any leading separators from
    while b[0] == "/" or b[0] == ".":
        b = b[1:]

    return a + b
